test/datasets_utils.py

```diff
@@ -572,9 +572,22 @@ def test_transforms_v2_wrapper(self, config):
 
         try:
             with self.create_dataset(config) as (dataset, _):
-                wrapped_dataset = wrap_dataset_for_transforms_v2(dataset)
-                wrapped_sample = wrapped_dataset[0]
-                assert tree_any(lambda item: isinstance(item, (Datapoint, PIL.Image.Image)), wrapped_sample)
+                for target_keys in [None, "all"]:
+                    wrapped_dataset = wrap_dataset_for_transforms_v2(dataset, target_keys=target_keys)
+                    wrapped_sample = wrapped_dataset[0]
+
+                    flat_sample, _ = tree_flatten(wrapped_sample)
+
+                    for item in flat_sample:
+                        if isinstance(item, (Datapoint, PIL.Image.Image)):
+                            break
+                        elif isinstance(item, torch.Tensor):
+                            raise AssertionError(
+                                "Found a plain tensor before an explicit image or video. "
+                                "This is not compatible with the tensor fallback heuristic."
+                            )
+                    else:
+                        raise AssertionError("Found no image or video in sample.")
```

Comment:
Driveby: instead of just checking that we have an explicit image, we now also check that no plain tensor comes before it.

Reply:
Ah shoot my bad. You are right. Let me revert.